### PR TITLE
Rename "legal text" to "warning text"

### PR DESCRIPTION
**Pull request by dashouse**
- Updates every reference to "legal text" to "warning text"
- Updates the template to take text or html
- A 1px change to icon position :)

https://trello.com/c/45Fnu1yr/254-2-update-legal-text-component-api-and-name

### CODE REVIEW

File: src/components/warning-text/_warning-text.scss [hunk 1]
```diff
@@ -26,14 +26,14 @@
 
     min-width: 38px;
     min-height: 35px;
-    margin-top: -19px; // Half the height of the 38px circle
+    margin-top: -20px; // Half the height of the 38px circle
```
Comment (nickcolley) [inline src/components/warning-text/_warning-text.scss]: Nit: This comment is confusing now, could update to say 'half the height of the circle'

**Comment by nickcolley** [comment]
@dashouse would you mind adding an entry for this to the unreleased section of the changelog? https://github.com/alphagov/govuk-frontend/blob/master/CHANGELOG.md#unreleased

Let me know if you want to pair on the versioning side of it.

**Comment by dashouse** [comment]
@igloosi There's no gotcha's hiding with renaming a component I haven't addressed is there?

**Comment by igloosi** [comment]
@dashouse not for this bit, just need to remember to update all.scss fiel in packages

did you rebase fro master?
it currently fails with 
```
Message:
    src/globals/scss/govuk-frontend.scss
Error: File to import not found or unreadable: ../../components/link/link.
       Parent style sheet: /Users/janikraner/Desktop/git/govuk-frontend/src/globals/scss/govuk-frontend.scss
        on line 19 of src/globals/scss/govuk-frontend.scss
>> @import "../../components/link/link";
```

**Comment by nickcolley** [comment]
@dashouse we will need make sure to follow step 11. in https://github.com/alphagov/govuk-frontend/blob/master/docs/publishing.md

**Comment by igloosi** [comment]
@nickcolley i have removed the link component reference from govuk-frontend.scss file

**Comment by nickcolley** [comment]
👍 when green

**Comment by hannalaakso** [comment]
@dashouse I've updated `CHANGELOG` to reflect changes in this PR. Could you please check I've captured everything.

Our new guidance for updating `CHANGELOG` is here, I've only just added it: https://github.com/alphagov/govuk-frontend/pull/433/files

**Comment by dashouse** [comment]
@igloosi @nickcolley @hannalaakso Sorry I wasn't around for most of this yesterday. Thanks for fixing :)

**Pull request merged**